Utilities/File.cpp

```diff
@@ -578,14 +580,19 @@ bool fs::copy_file(const std::string& from, const std::string& to, bool overwrit
 	}
 
 	// Here we use kernel-space copying for performance reasons
-#if defined(__APPLE__) || defined(__FreeBSD__)
-	// fcopyfile works on FreeBSD and OS X 10.5+ 
+#if defined(__APPLE__)
+	// fcopyfile works on OS X 10.5+
 	if (::fcopyfile(input, output, 0, COPYFILE_ALL))
-#else
+#elif defined(__DragonFly__) || defined(__FreeBSD__)
+	if (::sendfile(input, output, 0, 0, NULL, NULL, 0))
+#elif defined(__linux__) || defined(__sun)
 	// sendfile will work with non-socket output (i.e. regular file) on Linux 2.6.33+
 	off_t bytes_copied = 0;
 	struct ::stat fileinfo = { 0 };
 	if (::fstat(input, &fileinfo) || ::sendfile(output, input, &bytes_copied, fileinfo.st_size))
+#else // NetBSD, OpenBSD, etc.
+	errno = 0;
+	if (true)
```

Comment:
`#error` here maybe more informative but then why not just drop `fs::copy_file` due to lack of consumers.